common/logger/src/struct_log.rs

```diff
@@ -94,13 +94,19 @@ impl Write for TcpWriter {
         // Attempt to write, and if it fails clear underlying stream
         // This doesn't guarantee a message cut off mid send will work, but it does guarantee that
         // we will connect first
-        let mut stream = self.stream.as_ref().unwrap();
-        let result = stream.write(buf);
-        if result.is_err() {
+        if let Some(mut stream) = self.stream.as_ref() {
+            let result = stream.write(buf);
+            if result.is_err() {
+                self.stream = None;
+            }
+            result
+        } else {
             self.stream = None;
+            Err(io::Error::new(
+                io::ErrorKind::NotConnected,
+                "Can't write to strean, not connected",
```

Comment:
@gregnazario ?

Reply:
So, this was using the `Option<Stream>` as a poor man's lock on whether to refresh the connection or not.  This is used in a single threaded fashion, and I'm not sure that the situation here that seems to be concerned about can happen.

Also `s/strean/stream/g`

Reply:
> Please revert the changes which unwrap on lock poisoning per our [coding guidelines](https://github.com/libra/libra/blob/master/documentation/coding_guidelines.md#error-handling). Our codebase is not equipped to handle poisoned locks, and the only way a lock becomes poisoned is if a thread panics while holding the lock.

done

Reply:
even if the situation in question won't happen (panicking on this unwrap), we still want to enable the lint to block all usage of unwrap in functions of type result , so I think making the changes now before the lint becomes available is the right thing to do. We want to reach a baseline of 0 violations of the lint before updating the lint, so that we won't have to fix all the violations in 1 massive PR in the future.